### DISCUSSION TITLE
building sqlite3 with additional fts5 stemmers

### DISCUSSION THREAD

**Discussion by olastor** [jart/cosmopolitan#404] [Q&A]
I am trying to figure out how to integrate [this extension](https://github.com/abiliojr/fts5-snowball) for sqlite3 / fts5 for a custom build of redbean to have full text search with stemming for multiple languages after activating fts5 as described [here](https://github.com/jart/cosmopolitan/issues/216#issuecomment-884664757).

I tried building without `-DSQLITE_OMIT_LOAD_EXTENSION` ([source](https://github.com/jart/cosmopolitan/blob/b30f5c0c4fc0b334e2b27389bc09abe718e2a2ae/third_party/sqlite3/sqlite3.mk#L123)) in order to be able to later load the extension, but that gave me some errors: [sqlite_errors.txt](https://github.com/jart/cosmopolitan/files/8645135/sqlite_errors.txt). So I assume that is not meant to be activated.

My next approach would be to include the plugin's source code into the sqlite3 folder and then add the stemmer as new buitin stemmer, but I am stuck how I could compile that with the snowball submodule in there that has its own folder structure and complex Makefile.

Does anyone have any hint if and how that might be possible?

**Comment by pkulchenko** [comment]
This is not going to work out of the box, as cosmopolitan executables don't support dlopen. There have been several discussions on this topic in various tickets, so this may still come, but the easiest way is probably to compile this extension statically, if you can find a way to merge it with the rest of the SQLite code and initialize at the correct time. It seems like this may be supported by SQLite itself: [Automatically Load Statically Linked Extensions](https://www.sqlite.org/c3ref/auto_extension.html). [This SO answer](https://stackoverflow.com/questions/30898113/how-to-compile-an-extension-into-sqlite) may also be relevant.

**Reply by olastor** [reply]
Thanks! The extension itself file I'd need to compile with cosmopolitan somehow in order to sustain compatibility?

**Reply by pkulchenko** [reply]
Right; it would need to be compiled together with the rest of the SQLite code (using cosmopolitan lib) to form one executable
